File: 0007-reverse-integer/0007-reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var ans=0;
    while(x!=0){
         ans=parseInt(ans*10+x%10);
        console.log(ans); 
        x=parseInt(x/10);
        if(Math.abs(ans)>2147483647) return 0;
    }
    return ans;
    
};

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long ans=0;
        while(x!=0){
            ans=ans*10+x%10;
            cout<<ans<<endl;
            x/=10;
            if(abs(ans)>INT_MAX) return 0;
            
        }
        return ans;
        
    }
};